components/EmotionList.js

```diff
@@ -1,17 +1,40 @@
 import styled from "styled-components";
 import Link from "next/link";
 import EmotionCard from "./EmotionCard";
+import { useState } from "react";
+import { emotions } from "@/lib/emotions";
 
 export default function EmotionList({ objects }) {
+  const [selectedFilter, setSelectedFilter] = useState("");
+
+  const filteredObjects = selectedFilter
+    ? objects.filter((item) => item.emotion === selectedFilter)
+    : objects;
+
   return (
     <StyledEmotionList>
-      {objects.length === 0 ? (
+      <StyledHeadline>Filter emotion type</StyledHeadline>
+      <StyledDiv>
+        {emotions.map((emotion) => (
+          <StyledSpan
```

Comment:
Similar to the StyledDiv I suggest to name it `StyledSelectFilter`. When the project grows bigger there might be more div´s, span´s etc. so it might be good to specify them to better tell them apart later on.

Reply:
You are right. Just changed to StyledFilter :)